.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,41 @@
+name: Build and Test Spring Boot Application
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
```

Comment:
Should these be limited to certain branches instead of all of them? I think that building and testing only pushed/merges into dev and main would be enough.

Any other opinions?

Reply:
Well I thought that this would prevent broken changes to be merged. But yeah I could be limit this to main, dev and feature branches so work branches wouldn't need the run

Reply:
Sounds good. You're right that it could be smart to prevent broken merges. Just to keep in mind GitHub should have some sort of minute limitation for pipelines on the free tier which we probably don't want to exceed. :D

Reply:
https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions

2000 minutes on the free tier per month -> should be enough.

Reply:
Yes should be enough. Lets keep eye on this and reduce the amount of runs if we seem to run out of minutes :D